pkg/storage/batch.go

```diff
@@ -717,6 +717,10 @@ func fetchLazyChunks(ctx context.Context, s config.SchemaConfig, chunks []*LazyC
 			}
 			chks, err := fetcher.FetchChunks(ctx, chks, keys)
 			if err != nil {
+				if ctx.Err() != nil {
+					errChan <- nil
+					return
+				}
```

Comment:
Is `ctx.Err()` only `!= nil` if `err != nil`? Otherwise it would make sense to move it out of the `if err != nil {}` block?

Reply:
According to your tips, this PR code is clearer.